sirepo/pkcli/admin.py

```diff
@@ -55,20 +80,50 @@ def create_examples():
         with sirepo.auth_db.session_and_lock(), auth.set_user_outside_of_http_request(
             uid
         ):
-            for sim_type in feature_config.cfg().sim_types:
-                names = [
-                    x.name
-                    for x in simulation_db.iterate_simulation_datafiles(
-                        sim_type,
-                        simulation_db.process_simulation_list,
-                        {
-                            "simulation.isExample": True,
-                        },
-                    )
-                ]
-                for example in simulation_db.examples(sim_type):
-                    if example.models.simulation.name not in names:
-                        _create_example(example)
+            callback()
+
+
+def reset_examples():
+    def _reset():
+        c = []
+        r = []
+        for sim_type in feature_config.cfg().sim_types:
+            sims = [
+                x
+                for x in simulation_db.iterate_simulation_datafiles(
+                    sim_type,
+                    simulation_db.process_simulation_list,
+                    {"simulation.isExample": True},
+                )
+            ]
+            _build_delete_and_revert_lists(c, r, sims, sim_type)
+        _revert_sims(r)
+        _delete_sims(c)
+
+    _access_sim_db_and_callback(_reset)
+
+
+def _build_delete_and_revert_lists(delete_list, revert_list, simulations, sim_type):
```

Comment:
I don't like that they are modifying values in this particular way. I would say create a `ops = PKDict(delete=[], revert=[])` and pass ops to _build_ops (_build_delete_and_revert_lists is too complicated a name). Then rename functions _delete and _revert, which are all nested functions in reset.